src/app/system/manage-data-tables/edit-data-table/edit-data-table.component.html

```diff
@@ -0,0 +1,94 @@
+              Application Table Name is <strong>required</strong>
+            </mat-error>
+          </mat-form-field>
+
+          <div fxFlexFill fxLayout="row wrap" fxLayout.lt-md="column" formArrayName="columns"
+            *ngFor="let column of columns.controls; let i = index;">
+
+            <div fxFlexFill fxLayout="row wrap" fxLayoutGap="2%" fxLayout.lt-md="column" [formGroupName]="i">
+
+              <mat-form-field fxFlex="18%">
+                <mat-label>Column Name</mat-label>
+                <input matInput required formControlName="name">
+                <mat-error *ngIf="columns.at(i).controls.name.hasError('required')">
+                  Column Name is <strong>required</strong>
+                </mat-error>
+              </mat-form-field>
+
+              <mat-form-field fxFlex="16%">
+                <mat-label>Column Type</mat-label>
+                <mat-select required formControlName="type">
+                  <mat-option *ngFor="let columnType of columnTypeData" [value]="columnType.value">
+                    {{ columnType.displayValue }}
+                  </mat-option>
+                </mat-select>
+              </mat-form-field>
+
+              <div fxFlex="13%" class="mandatory-wrapper">
+                <mat-checkbox labelPosition="before" formControlName="mandatory" class="mandatory">Mandatory</mat-checkbox>
+              </div>
+
+              <mat-form-field fxFlex="16%">
+                <mat-label>Column Length</mat-label>
+                <input matInput required type="number" formControlName="length">
+              </mat-form-field>
+
+              <mat-form-field fxFlex="16%">
+                <mat-label>Column Code</mat-label>
+                <mat-select required formControlName="code">
+                  <mat-option *ngFor="let columnCode of columnCodeData" [value]="columnCode.name">
+                    {{ columnCode.name }}
+                  </mat-option>
+                </mat-select>
+              </mat-form-field>
+
+              <span fxFlex>
+                <button *ngIf="columns.length !== 1" type="button" mat-icon-button (click)="removeColumn(columns, i)">
+                  <fa-icon icon="minus-circle" size="lg"></fa-icon>
+                </button>
+                <button *ngIf="i === 0" type="button" mat-icon-button color="primary" (click)="addColumn(columns)">
+                  <fa-icon icon="plus-circle" size="lg"></fa-icon>
+                </button>
+              </span>
+
+            </div>
+
+          </div>
+
+        </div>
+
+      </mat-card-content>
+
+      <mat-card-actions fxLayout="row" fxLayout.xs="column" fxLayoutAlign="center" fxLayoutGap="5px">
+        <button type="button" mat-raised-button [routerLink]="['../']">Cancel</button>
+        <button mat-raised-button color="primary" [disabled]="!dataTableForm.valid || dataTableForm.pristine && !dataTableChangesData.dropColumns">Submit</button>
```

Comment:
```suggestion
        <button mat-raised-button color="primary" [disabled]="!dataTableForm.valid || dataTableForm.pristine">Submit</button>
```

We can't follow this approach. Disabling of rows in case of removal doesn't look good to me from the user's perspective. It will create confusion. You'll have to figure out another way.

Reply:
I can't really see another way, as you mentioned above that you were thinking to table approach but that (in my opinion) looks more confusing to me.

Reply:
We will reiterate over that for this component later. 
First, here instead of disabling the row, let us try to remove the row as you were doing earlier. Then resolve the editing issue. How is disabling solving it?

Reply:
In the beginning, we are subscribing to all the columns but when a user removes one, we aren't un-subscribing all and then again subscribing in the correct order (It will take loads of code + it is inefficient) whereas when I disable it, it automatically gets unsubscribed and the order is also maintained.

Reply:
Why not unsubscribe the dropped column only?